lib/linter-views.coffee

```diff
@@ -115,11 +116,11 @@ class LinterViews
   updateBubble: (point) ->
     @removeBubble()
     return unless @showBubble
-    return unless @messages.size
+    return unless @messages.length
     activeEditor = atom.workspace.getActiveTextEditor()
     return unless activeEditor?.getPath()
     point = point || activeEditor.getCursorBufferPosition()
-    try @messages.forEach (message) =>
+    try @messages.toArray().forEach (message) =>
```

Comment:
these are the only things I don't really like, they could be replaced with comprehensions:
```coffee
try
  for index in [0..@messages.length - 1]
    message = @messages.get(index)
    continue unless message.currentFile
    continue unless message.range?.containsPoint point
    @bubble = activeEditor.markBufferRange([point, point], {invalidate: 'never'})
    activeEditor.decorateMarker(
      @bubble
      {
        type: 'overlay',
        position: 'tail',
        item: @renderBubble(message)
      }
    )
    throw null
```

That would be faster since otherwise *deque* would always create a new array in `toArray`.
Another option would be to monkey patch *deque* to support `forEach`.